crates/wasmtime/src/runtime/gc/enabled/rooting.rs

```diff
@@ -186,12 +186,19 @@ pub(crate) use sealed::*;
 // Just `pub` to avoid `warn(private_interfaces)` in public APIs, which we can't
 // `allow(...)` on our MSRV yet.
 #[doc(hidden)]
+#[repr(C)] // NB: if this layout changes be sure to change the C API as well
 pub struct GcRootIndex {
     store_id: StoreId,
     generation: u32,
     index: PackedIndex,
 }
 
+const _: () = {
+    // NB: these match the C API which should also be updated if this changes
+    assert!(std::mem::size_of::<GcRootIndex>() == 16);
+    assert!(std::mem::align_of::<GcRootIndex>() == 8);
+};
```

Comment:
Maybe these asserts should be for `ManuallyRooted` rather than `GcRootIndex`, just so that if we add a field to the former, but leave the latter as-is, we get build errors because that mismatch would introduce unsafety.